client/components/autocomplete2/test/index.test.js

```diff
@@ -47,12 +48,25 @@ describe('Autocomplete', () => {
     expect(component.instance().engine.get.mock.calls[0][0]).toBe('a title')
   })
 
-  it('OnSelect calls props.onSelect with selected id', () => {
+  it('OnSelect calls props.onSelect with selected id', async () => {
     const component = mount(
       <Autocomplete {...props} />
     )
-    component.instance().onSelect(results[0])
-    expect(props.onSelect.mock.calls[0][0][0]).toBe(results[0]._id)
+    await component.instance().onSelect(searchResults[0])
+    expect(props.onSelect.mock.calls[0][0][0]).toBe(searchResults[0].id)
+  })
+
+  it('Returns a custom resObject on select if provided', async () => {
+    const formatSelected = (item) => {
+      return new Article(item)
```

Comment:
I think I might be a little confused by this -- in order to test the code you wrote, I would think this function needed to return a Promise. If this is the way we intend to write it, perhaps we don't need async/await at all?

Reply:
It doesn't wait for component.instance().onSelect (which is also async) to complete before hitting the expect if I don't include async in the test as well.  Async is a new concept to me so perhaps I am doing it wrong?

Reply:
Oh i misunderstood your comment -- yes can make the sample prop async too

Reply:
I think it depends on how we intend to use `formatSelected`. If `formatSelected` is meant to format objects by wrapping it in an `Article` and if it's not intended to make asynchronous requests, then we shouldn't need to use async/await in this PR. The async/await stuff comes in handy when we're dealing with Promises or when we want to avoid callbacks by turning things into Promises. 

For example, the `onSelect` could look like:

```
onSelect = (selected) => {
   const { items, onSelect, formatSelected } = this.props
   const item = formatSelected ? formatSelected(selected) : selected.id

   items.push(item)
   ...
}
```

Reply:
we do need it -- was implementing for fetching artworks -- new test coming soon